pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -269,6 +269,9 @@ func (am *Alertmanager) SaveAndApplyDefaultConfig(ctx context.Context) error {
 		if err := am.applyConfig(cfg, []byte(am.Settings.UnifiedAlerting.DefaultConfiguration)); err != nil {
 			return err
 		}
+		if err := am.Store.MarkAlertmanagerConfigurationAsSuccessful(ctx, cmd.ResultID); err != nil {
```

Comment:
Shouldn't this also be applied in `SaveAndApplyConfig` below?

In fact, I think it might be best to remove these spread out calls and add a single centralized call of `MarkAlertmanagerConfigurationAsSuccessful` directly in `Alertmanager.ApplyConfig` (this might require fiddling with uses of `Alertmanager.applyConfig` to instead use `Alertmanager.ApplyConfig` instead). This last part isn't strictly necessary if it proves unwieldy but it would be a good thing IMO.